cyhy_report/customer/generate_report.py

```diff
@@ -2124,6 +2173,49 @@ def __table_vulnerability_deltas(self):
         self.__results["resolved_vulnerabilities"] = d_resolved_vulns
         self.__results["resolved_vulnerability_counts"] = d_resolved_counts
 
+        # Calculate Known Exploited Vulnerability (KEV) Counts
+        # Active KEV counts and find maximum active KEV age
+        active_kev_counts = Series([0, 0, 0, 0])
+        if len(df0):
+            df_active_kev = df0[df0["kev"] == True]
+
+            kev_max_age = 0
+            if len(df_active_kev):
+                kev_max_age = df_active_kev["age"].max()
+                active_kev_counts = df_active_kev.groupby(
+                    "severity"
+                ).size()
+        active_kev_counts = active_kev_counts.reindex_axis([4, 3, 2, 1]).fillna(0)
+        active_kev_counts = active_kev_counts.apply(np.int)
+        d_active_kev_counts = active_kev_counts.to_dict()
+        d_active_kev_counts = self.__level_keys_to_text(d_active_kev_counts, lowercase=True)
+        self.__results["active_kev_counts"] = d_active_kev_counts
+        self.__results["active_kev_max_age"] = kev_max_age
```

Comment:
I'm not very familiar with Pandas (or whatever it is you're using) so some comments explaining in generic mathematical language what is going on at each stage would be helpful.

It looks like you're creating a 4-vector of KEV counts by severity (which you reverse the sorting on via the `reindex-axis` call) from `df0`, but I'm unclear what that poorly-named variable represents.  I assume "df" corresponds to "data fields".

It looks like this jazz is mostly repeated from above, so you could just explain it once the first time it takes place and refer the reader there with a brief comment.

Reply:
`df` is shorthand for a [pandas.DataFrame](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html).  I added comments in https://github.com/cisagov/cyhy-reports/pull/71/commits/4d255bd4ef7b48c564bfb8cc61086e5b69f09e67, so let me know if that works for you or if anything needs additional clarification.

Reply:
Perfect!